src/modelutils.jl

```diff
@@ -981,3 +981,135 @@ function topicdist(model::TopicModel, doc_indices::Vector{<:Integer})
+
+	## Collect confirmation scores in a results channel
+	coherence_values = Channel{Float64}(buffer) do ch
+        Threads.foreach(topic_word_pairs) do pair
+            confirmation = calculate_confirmation(pair, model.corp)
+            put!(ch, confirmation)
+        end
+    end
+    
+	## Accumulate results channel
+    sum_coherence = 0.0
+    coherence_val_count = 0
+	num_pairs = 0
+    for r in coherence_values
+		num_pairs += 1
+		if isnan(r)
+			continue
+		else
+			sum_coherence += r
+			coherence_val_count += 1
+		end
+    end
+
+	(num_pairs > 0) || throw(error("None of the topic words appear in the vocabulary"))
+	(coherence_val_count > 0) || throw(error("None of the topic words appear in the coprus"))
+
+    coherence = sum_coherence / coherence_val_count
+
+    return coherence
+end
+
+function one2prev_generator(topic_words::Matrix{Int}, vocab::Dict{Int, String}, buffer=Threads.nthreads())
+	"Create a channel the returns one topic word pair at a time."
+
+    Channel{Tuple{Int, Int}}(buffer) do ch
+		for i in CartesianIndices(topic_words)
+			row, col = i[1], i[2]
+			current_tok = topic_words[i]
+
+			## Skip this topic word if it is not found in the vocabulary
+			current_gram = if current_tok in keys(vocab)
+				vocab[current_tok]
+			else
+				continue
+			end
+
+			for j in 1:row
+				previous_tok = topic_words[j,col]
+				gram_overlap::Bool = false
+				
+				## Skip this topic word pairing if previous is not found in the vocabulary
+				previous_gram = if previous_tok in keys(vocab)
+					vocab[previous_tok]
+				else
+					continue
+				end
+
+				## If one token is a n-gram, split them and check if either token
+				## Is completely contained in the other token
+				if occursin(" ", current_gram*previous_gram)
+					cur_gram_split = split(current_gram, " ")
+					prev_gram_split = split(previous_gram, " ")
+					gram_overlap = 
+					sum([1 for w in cur_gram_split if w in prev_gram_split]) == length(cur_gram_split) ||
+					sum([1 for w in prev_gram_split if w in cur_gram_split]) == length(previous_gram)
+				end
+
+				if !(gram_overlap)
+					put!(ch, (current_tok, previous_tok))
+				end
+			end
+        end
+    end
+end
+
+function calculate_confirmation(word_pair::Tuple{Int, Int}, corp::Corpus)
+	"Find the relationship between this word pair across all documents."
+
+    ## Count how many documents contain one word, the other, or both
+    num_w_current = 0
+    num_wo_current = 0
+    num_w_prev_not_current = 0
+    num_w_pair = 0
+    for doc in corp
+        terms = doc.terms
+        if word_pair[1] in terms
+            num_w_current += 1
+            if word_pair[2] in terms
+                num_w_pair += 1
+            end
+        else 
+            num_wo_current += 1
+            if word_pair[2] in terms
+                num_w_prev_not_current += 1
+            end
+        end
+    end
+    
+	## If either word is completely missing from the corpus, then we skip this confirmation
```

Comment:
Return a Nan if either word is totally missing from the corpus so that we can skip this pair when we accumulate at line 1016.  If they're all missing (maybe because the corpus is empty), then we raise another error at line 1027.

Reply:
I think Nan is better than zero because just because a word is missing from the corpus doesn't mean it's unrelated to the the other in its pair. Could just be a rare word that didn't make it into the corpus.